client/src/js/modules/summary/views/summary.vue

```diff
@@ -0,0 +1,1775 @@
+                Object.keys(data['0']),
+                // values
+                ...Object.values(data).map(item => Object.values(item))
+            ]
+                .reduce((string, item) => {
+                    string += item.join(',') + '\n';
+                    return string;
+            }, '');
+
+            return result;
+        },
+        exportCSV(csv) {
+
+            const anchor = document.createElement('a');
+            anchor.href = 'data:text/csv;charset=utf-8,' + encodeURIComponent(csv);
+            anchor.target = '_blank';
+            anchor.download = 'summary_export.csv';
+            anchor.click();
+
+        },
+        getProcRow(result, field, type) {
+            // return an array for results, also change decimal places for floats
+                if (result[field]) {
+
+                    var dataArray = result[field].split(",");
+
+                    if( type !== 'text') {
+                        if(type == 'decimal4') {
+                            return dataArray.map(function (x) { 
+                                return parseFloat(x).toFixed(4);
+                             });
+                        } else {
+                            return dataArray.map(function (x) { 
+                                return parseFloat(x).toFixed(2);
+                             });
+                        }
+                    };
+                        
+
+                    return dataArray;
+                }
+                else {
+                    return [];
+                };
+        },
+        checkedColumns(index) {
+            // add columns to selected or deselected object for pill display
+            this.summaryParameters[index].checked = !this.summaryParameters[index].checked;
+
+            if (this.summaryParameters[index].checked == false){
+
+            this.deselectedColumns.push(this.summaryParameters[index]);
+            this.selectedColumns = this.selectedColumns.filter(item => item !== this.summaryParameters[index]);
+            }
+            else {
+            this.selectedColumns.push(this.summaryParameters[index]);
+            this.deselectedColumns = this.deselectedColumns.filter(item => item !== this.summaryParameters[index]);
+
+            }
+
+
+        },
+        resizePills() {
+            // change number of pills visible based on width of window
+
+            if (this.windowWidth > 1000 ) {
+                this.pillIndex = 1;
+            }
+
+            if (this.windowWidth > 1200 ) {
+                this.pillIndex = 2;
+            }
+
+            if (this.windowWidth > 1600 ) {
+                this.pillIndex = 3;
+            }
+
+            if (this.windowWidth > 1800 ) {
+                this.pillIndex = 4;
+            }
+
+            if (this.windowWidth > 2200 ) {
+                this.pillIndex = 5;
+            }
+
+            if (this.windowWidth > 2800 ) {
+                this.pillIndex = 6;
+            }
```

Comment:
Could we reverse the order of these and make them else-ifs?

Reply:
You can fit these to a function if you make the values more homogeneous, or possibly even use CSS to do this

Reply:
Changed to this approach which seems to work and is cleaner:

```
 if (this.windowWidth > 1000 & this.windowWidth < 2900) {

                const windWidthRounded = Math.ceil(this.windowWidth/100)*100;

                if( windWidthRounded % 200 === 0) {

                    this.pillIndex = (windWidthRounded - 1000)/200;

                }

            } 
```